src/scala/scripts/PBGenerateRequest.scala

```diff
@@ -6,16 +6,44 @@ case class PBGenerateRequest(jarOutput: String, scalaPBOutput: Path, scalaPBArgs
 
 object PBGenerateRequest {
 
-  def from(args: java.util.List[String]): PBGenerateRequest = {
+  // If there is a virtual import in the path, then
+  // the path under there is pretty friendly already to protoc
+  def stripToVirtual(e: String): String = {
+    val v = "/_virtual_imports/"
+    val idx = e.indexOf(v)
+    if(idx >= 0) {
+      e.drop(idx + v.size)
+    } else e
+  }
+
+  def from(tmpDir: Path)(args: java.util.List[String]): PBGenerateRequest = {
     val jarOutput = args.get(0)
     val protoFiles = args.get(4).split(':')
-    val includedProto = args.get(1).drop(1).split(':').distinct.map { e =>
-      val p = e.split(',')
-      // If its an empty string then it means we are local to the current repo for the key, no op
-      (Some(p(0)).filter(_.nonEmpty), p(1))
-    }.collect {
-      // if the to compile files contains this absolute path then we are compiling it and shoudln't try move it around(duplicate files.)
-      case (Some(k), v) if !protoFiles.contains(v) => (Paths.get(k), Paths.get(v))
+    val protoFilesToBuild = protoFiles.map { e => s"${tmpDir.toFile.toString}/${stripToVirtual(e)}"}
+
+    val includedProtoSplit: List[(String, String)] = args.get(1).drop(1).split(':').map { e =>
+      val arr = e.split(',')
```

Comment:
want to add a `require(arr.length == 2, s"expected exactly two comma separated items in ${arr.toList}")` just to guard against anything funny changing later?